### ISSUE TITLE
Calendar - model not updated on time change

### ISSUE THREAD

**Issue by britvik** [primefaces/primeng#2446]
**I'm submitting a ...**
```
[X] bug report
```

**Current behavior**
Model is not updated when time changes.

**Expected behavior**
Model is updated on time change.

**Minimal reproduction of the problem with instructions**
1. Open primeNG [calendar demo page](https://www.primefaces.org/primeng/#/calendar).
2. Click on Time input (one with date **and time** calendar)
3. Hit arrow to change time multiple times.
4. Watch how model changes only once.

**Please tell us about your environment:**
* **Angular version:** 4.0.0
* **PrimeNG version:** 4.0.0-rc.1
* **Browser:** [all]
* **Language:** [all]

**Comment by britvik** [comment]
I believe the problem is, that you change time portion inside existing date object and angular doesn't detect this change. If you construct new Date object on time change, angular will detect it.

**Comment by VagrantAI-c** [comment]
Expirenced the same problem, solved it with changing `(ngModelChange)` to `(onSelect)` in my template

**Comment by cagataycivici** [comment]
Glad to hear issue is resolved.

**Comment by demeatha** [comment]
Still on primeng version 7 issue exists. 
Tried both approaches mentioned above. 
I used onSelect event, and I cloned event date value object, on each change.
Although, when date change comes from a wrapping component as input to p-calendar, then change detection doesn't trigger into that component( I guess issue is related with ControlValueAccessor implementation).
The only think that worked in my case was to get p-calendar reference and use it as a plain js object by calling function updateUI inside wrapper's ngOnChanges hook.
